system_test/aest_hard_fork_SUITE.erl

```diff
@@ -445,6 +471,123 @@ new_node_can_mine_on_old_chain_using_new_protocol(Cfg) ->
+    start_node(new_node1, Cfg),
+    {ok, {TopHash, TopHeight}} = restore_db_backup_on_node(new_node1, Cfg),
+    B1 = get_block_by_height(new_node1, TopHeight, Cfg),
+    ?assertEqual(TopHash, maps:get(hash, B1)),
+    %% Make sure balance of new_node1 is 0.
+    PubKey1 = get_public_key(new_node1, Cfg),
+    Balance1 = get_balance(new_node1, PubKey1, Cfg),
+    ?assertEqual(0, Balance1),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node1, PubKey1, Balance1]),
+    %% Sync the chain on new_node2 from new_node1.
+    start_node(new_node2, Cfg),
+    wait_for_height_syncing(TopHeight, [new_node2], {{45000, ms}, {200, blocks}}, Cfg),
+    B2 = get_block_by_height(new_node2, TopHeight, Cfg),
+    ?assertEqual(TopHash, maps:get(hash, B2)),
+    %% Make sure balance of new_node2 is 0.
+    PubKey2 = get_public_key(new_node2, Cfg),
+    Balance2 = get_balance(new_node2, PubKey2, Cfg),
+    ?assertEqual(0, Balance2),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node2, PubKey2, Balance2]),
+    %% Start mining on new_node2.
+    ok = mock_pow_on_node(new_node1, Cfg),
+    ok = mock_pow_on_node(new_node2, Cfg),
+    run_erl_cmd_on_node(new_node2, "aec_conductor:start_mining().", "ok", Cfg),
+    %% Check the last block of old protocol has genesis version.
+    wait_for_height_syncing(LastHeightOfOldProtocol, [new_node2], {{10000, ms}, {1000, blocks}}, Cfg),
+    B2OldProtocol = get_block_by_height(new_node2, LastHeightOfOldProtocol, Cfg),
+    ?assertEqual(9, maps:get(version, B2OldProtocol)),
+    %% Check the first block of new protocol has version of the new protocol.
+    wait_for_height_syncing(HeightOfNewProtocol, [new_node2], {{10000, ms}, {1000, blocks}}, Cfg),
+    B2NewProtocol = get_block_by_height(new_node2, HeightOfNewProtocol, Cfg),
+    ?assertEqual(10, maps:get(version, B2NewProtocol)),
+    %% Check that new_node2 got reward for mining.
+    MinedReward = 100,
+    aest_nodes:wait_for_value({balance, PubKey2, MinedReward}, [new_node2], 10000, Cfg),
+    MinedBalance = get_balance(new_node2, PubKey2, Cfg),
+    ?assertMatch(X when X >= MinedReward, MinedBalance),
+    ct:log("Mined balance on ~p with public key ~p is ~p", [new_node2, PubKey2, MinedBalance]),
+    %% Send spend transaction from new_node2 to new_node1.
+    Fee = 5,
+    BalanceToSpend = MinedBalance - Fee,
+    ok = post_spend_tx(new_node2, PubKey1, BalanceToSpend, Fee, Cfg),
+    ct:log("Sent spend tx with balance of ~p from node ~p with public key ~p to node ~p with public key ~p",
+           [BalanceToSpend, new_node2, PubKey2, new_node1, PubKey1]),
+    %% Make sure new_node1 received the balance.
+    aest_nodes:wait_for_value({balance, PubKey1, BalanceToSpend}, [new_node2], 10000, Cfg),
+    ReceivedBalance = get_balance(new_node2, PubKey1, Cfg),
+    ct:log("Balance of node ~p with public key ~p is ~p", [new_node1, PubKey1, ReceivedBalance]),
+    ?assertMatch(BalanceToSpend, ReceivedBalance),
+    ok.
```

Comment:
This test is incomplete as it does not check that distinct node can sync block with spend tx. (Same as https://github.com/aeternity/epoch/pull/978/files#r181797787 )